modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/loader/xml/TlsEnabledComponentUtils.java

```diff
@@ -111,6 +120,29 @@ private static boolean hasTlsContextFactoryParameter(ParameterizedModel model) {
     return model.getAllParameterModels().stream().anyMatch(TlsEnabledComponentUtils::isTlsContextFactoryParameter);
   }
 
+  private static boolean requiresTlsContextFactoryParameter(ParameterizedModel model) {
+    return model.getAllParameterModels().stream()
+        .filter(ParameterModel::isRequired)
+        .anyMatch(TlsEnabledComponentUtils::isTlsContextFactoryParameter);
+  }
+
+  private static boolean isTlsConfigurationRequired(ComponentAst componentAst) {
```

Comment:
this method name is confusing. If this is to check if the paramater is requried, why mentioning the `Configuration` term? When should I use  this method instead of `requiresTlsContextFactoryParameter`

Reply:
Yes, you are right, we use the Configuration term for other things already and it may be confusing. I was thinking about the settings for the TLS context, not a Configuration in the sense of an Extension's Configuration.

In any case, I think this method could very well be an overload of `requiresTlsContextFactoryParameter`. One operates on a `ComponentAst` and the other one operates on a `ParameterizedModel`. Also one already delegates on the other. How does it sound?